### PR TITLE
[#1040] rework schema definitions in geo processors

### PR DESCRIPTION
**Pull request by flomickl**
### Purpose
#1040  improve naming of runtime names and add missing epsg field in trajectory

### Remarks

PR introduces (a) breaking change(s): yes
tested with all changes according to 
https://cwiki.apache.org/confluence/x/QZBbDg
https://cwiki.apache.org/confluence/x/RpBbDg
https://cwiki.apache.org/confluence/x/SZBbDg
https://cwiki.apache.org/confluence/x/U5BbDg
https://cwiki.apache.org/confluence/x/XJBbDg

PR introduces (a) deprecation(s): no

### Breaking changes
* `LatLngToJtsPointProcessor`: Rename schema field `WKT_RUNTIME` to `GEOMETRY_RUNTIME`
* `ReprojectionProcessor`: Rename schema field `GEOM_RUNTIME` to `GEOMETRY_RUNTIME`
* `TrajectoryFromPointsProcessor`: Introduce schema field `TRAJECTORY_GEOMETRY_RUNTIME` and rename `TRAJECTORY_RUNTIME` to  `TRAJECTORY_EPSG_RUNTIME`
* `HaversineDistanceCalculatorProcessor`: Rename schema field `CALCULATED_DISTANCE_KEY` to `DISTANCE_RUNTIME_NAME`
* `SpeedCalculatorProcessor`: Rename schema field `SPEED_KEY` to `SPEED_RUNTIME_NAME`

### CODE REVIEW

**Review by dominikriemer — APPROVED** [review]

File: streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/latlngtojtspoint/LatLngToJtsPointProcessor.java [hunk 1]
```diff
@@ -101,7 +101,11 @@ public void onEvent(Event event, SpOutputCollector collector) throws SpRuntimeEx
     Point geom = SpGeometryBuilder.createSPGeom(lng, lat, epsg);
 
     if (!geom.isEmpty()) {
-      event.addField(WKT_RUNTIME, geom.toString());
+      // if activated the stream fails at all
+      //event.removeFieldBySelector(latitudeMapper);
+      //event.removeFieldBySelector(longitudeMapper);
```
Comment (bossenti) [inline streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/latlngtojtspoint/LatLngToJtsPointProcessor.java:106]: Can we please refrain from committing dead code? :)

File: streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/trajectory/TrajectoryFromPointsProcessor.java [hunk 2]
```diff
@@ -51,8 +51,9 @@ public class TrajectoryFromPointsProcessor extends StreamPipesDataProcessor {
   private static final String SUBPOINTS_KEY = "subpoints-key";
   private static final String DESCRIPTION_KEY = "description-key";
   private static final String TRAJECTORY_KEY = "trajectory-key";
-  private static final String TRAJECTORY_RUNTIME = "trajectoryWKT";
-  private static final String DESCRIPTION_RUNTIME = "trajectoryDescription";
+  private static final String TRAJECTORY_GEOMETRY_RUNTIME = "trajectory-geometry";
+  private static final String TRAJECTORY_EPSG_RUNTIME = "trajectory-epsg";
+  private static final String DESCRIPTION_RUNTIME = "trajectory-description";
```
Comment (bossenti) [inline streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/trajectory/TrajectoryFromPointsProcessor.java:56]: Do you not want to provide labels via the `strings.en` file?

**Review by bossenti — CHANGES_REQUESTED** [review]

File: streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/trajectory/TrajectoryFromPointsProcessor.java (hunk 2 above)
Comment (flomickl) [reply streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/trajectory/TrajectoryFromPointsProcessor.java:56]: @bossenti 
so far I never implemented the runtime names to the string.en file
What would be the benefit of it?


All necessary descriptions for the user are already written down:
traj_elements

Comment (bossenti) [reply streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/trajectory/TrajectoryFromPointsProcessor.java:56]: Ah sorry, my bad 🙈
Messed up the variables

File: streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/latlngtojtspoint/LatLngToJtsPointProcessor.java (hunk 1 above)
Comment (flomickl) [reply streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/latlngtojtspoint/LatLngToJtsPointProcessor.java:106]: it's not dead, it is in discussion :)
https://github.com/apache/streampipes/discussions/1276
it is not working currently but should be part of the processor

Comment (bossenti) [reply streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/latlngtojtspoint/LatLngToJtsPointProcessor.java:106]: Well then this PR isn't ready to be merged right?

Comment (flomickl) [reply streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/latlngtojtspoint/LatLngToJtsPointProcessor.java:106]: It is ready to merge. The pull request is for renaming the runtime names.
This code snipped is just a reference for the discussion.
Everything else is still working like before.
It is only a nice to have implementation but not required and can be resolved in a new PR after the discussion is finished :)

Comment (bossenti) [reply streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/latlngtojtspoint/LatLngToJtsPointProcessor.java:106]: Frankly speaking, I don't think it's a good idea to commit commented code for several reasons.
Alternatively, if you want to have a reference for your discussion you could create a gist, simply paste the code snippet into the discussion, or create a dedicate branch for it.
But I don't want to get too hung up on two lines of code here right now ✌🏼 
So if you want to do it that way, feel free 🙂

Comment (flomickl) [reply streampipes-extensions/streampipes-processors-geo-jvm/src/main/java/org/apache/streampipes/processors/geo/jvm/jts/processor/latlngtojtspoint/LatLngToJtsPointProcessor.java]: Yes, it's good that you care so much about clean code. I have removed it.

**Comment by flomickl** [comment]
@bossenti ok to merge?

**Review by bossenti — APPROVED** [review]

**Comment by bossenti** [comment]
Sure, just go ahead :)

**Pull request merged**